ut/alarm_tables_testing.cpp

```diff
@@ -0,0 +1,321 @@
+
+TEST_F(SNMPTest, ModelTable)
+{ 
+  // Create a custom alarm definition
+  AlarmDef::SeverityDetails cleared(AlarmDef::CLEARED, "Test alarm cleared description", "Test alarm cleared details");
+  AlarmDef::SeverityDetails raised(AlarmDef::CRITICAL, "Test alarm raised description", "Test alarm raised details");
+  AlarmDef::AlarmDefinition example(6666, AlarmDef::SOFTWARE_ERROR, {cleared, raised});
+  AlarmTableDef def_cleared(example, cleared);
+  AlarmTableDef def_raised(example, raised);
+  AlarmTableDefs::get_instance().insert_def(def_cleared);
+  AlarmTableDefs::get_instance().insert_def(def_raised);
+
+  init_alarmModelTable();
+  
+  // Verifies the above alarm data is exactly what is stored in
+  // AlarmModelTable by querying the sixth column in alarmModelTable which
+  // should contain the description of the alarm)
+  ASSERT_STREQ("\"Test alarm cleared description\"\n", snmp_get_raw(alarmModelTable_entry + ".6.0.6666.1", buf, sizeof(buf)));
+
+  //Here we use AlarmModelState (6) to query AlarmModelTable, for details of the
+  //mapping to ituAlarmPerceivedSeverity see
+  //https://tools.ietf.org/html/rfc3877#section-5.4
+  ASSERT_STREQ("\"Test alarm raised description\"\n", snmp_get_raw(alarmModelTable_entry + ".6.0.6666.6", buf, sizeof(buf)));
+
+  // Verifies the third column (which contains the notification type) is the
+  // alarmClearState oid and the alarmActiveState oid for the cleared and raised
+  // alarm respectively.
+  ASSERT_STREQ("iso.3.6.1.2.1.118.0.3\n", snmp_get_raw(alarmModelTable_entry + ".3.0.6666.1", buf, sizeof(buf)));
+  ASSERT_STREQ("iso.3.6.1.2.1.118.0.2\n", snmp_get_raw(alarmModelTable_entry + ".3.0.6666.6", buf, sizeof(buf)));
+}
+
+TEST_F(SNMPTest, ituAlarmTable)
+{
+  // Creates a custom alarm definition
+  AlarmDef::SeverityDetails cleared(AlarmDef::CLEARED, "Test alarm cleared description", "Test alarm cleared details");
+  AlarmDef::SeverityDetails raised(AlarmDef::CRITICAL, "Test alarm raised description", "Test alarm raised details");
+  AlarmDef::AlarmDefinition example(6666, AlarmDef::SOFTWARE_ERROR, {cleared, raised});
+  AlarmTableDef def_cleared(example, cleared);
+  AlarmTableDef def_raised(example, raised);
+  AlarmTableDefs::get_instance().insert_def(def_cleared);
+  AlarmTableDefs::get_instance().insert_def(def_raised);
+  
+  init_ituAlarmTable();
+
+  ASSERT_STREQ("\"Test alarm cleared details\"\n", snmp_get_raw(ituAlarmTable_entry + ".4.0.6666.1", buf, sizeof(buf)));
+  //Here we use ituAlarmPerceivedSeverity (3) to query ituAlarmTable, for
+  //details of the mapping to AlarmModelState see
+  //https://tools.ietf.org/html/rfc3877#section-5.4
+  ASSERT_STREQ("\"Test alarm raised details\"\n", snmp_get_raw(ituAlarmTable_entry + ".4.0.6666.3", buf, sizeof(buf)));
+  ASSERT_EQ(163, snmp_get(ituAlarmTable_entry + ".3.0.6666.1"));
+  ASSERT_EQ(163, snmp_get(ituAlarmTable_entry + ".3.0.6666.3"));
+}
+
+TEST_F(SNMPTest, ActiveTableMultipleAlarms)
+{
+  init_alarmActiveTable("10.154.153.133");
```

Comment:
You flagged that this didn't seem optimal, and you're right - for example, running just the ActiveTableClearingUnraisedAlarm test won't work.

I think there are two possibilities:

- create a new test subclass with a SetUpTestCase function that calls init_alarmActiveTable, so it will run once before any of these tests that require it. You could also have a TearDown function calling `alarmActiveTable_trap_handler(defN_cleared);` for each definition, in case any test fails to clean up its alarms.
- define a destroy_activeAlarmTable method that resets everything to a clean state, then create a new test subclass with SetUp and TearDown functions that call init_alarmActiveTable and destroy_activeAlarmTable respectively, so that each test gets a clean state.

The latter is a bit cleaner but more work.